File: lib/to_mdx/parsers.rb
# frozen_string_literal: true

require_relative "./parsers/open_document"

module ToMdx
  module Parsers
    def self.parse(file_path, context)
      if file_path.end_with?(".odp")
        OpenDocument.parse(file_path, context)
      else
        raise "Unknown file type: #{file_path}"
      end
    end
  end
end

File: src/parsers.ts
import { parse as parseKeynote } from "./parsers/keynote/index.ts";
import { parse as parseOpenDocument } from "./parsers/open_document.ts";

export interface Options {
  useHeuristics?: boolean;
  useLlmDetection?: boolean;
}

export async function parse(outputRoot: string, filePath: string, options: Options): Promise<void> {
  if (filePath.endsWith(".odp")) {
    await parseOpenDocument(outputRoot, filePath, options);
  } else if (filePath.endsWith(".key")) {
    await parseKeynote(outputRoot, filePath, options);
  } else {
    throw new Error(`Unknown file type: ${filePath}`);
  }
}
